src/ray/gcs/redis_context.cc

```diff
@@ -211,12 +218,21 @@ Status ConnectWithRetries(const std::string &address, int port,
     *context = connect_function(address.c_str(), port);
     connection_attempts += 1;
   }
+  if (is_async_context) {
+    redisAsyncContext *async_context = reinterpret_cast<redisAsyncContext *>(*context);
+    async_context->data = redis_context;
+    redisAsyncSetDisconnectCallback(async_context, RedisAsyncContextDisconnectCallback);
+  }
```

Comment:
Looks good to me.
But I think it's better call redisAsyncSetDisconnectCallback in method Connect.
`RAY_CHECK_OK(ConnectWithRetries(address, port, redisAsyncConnect, &async_context_));`
`RAY_CHECK_OK(AuthenticateRedis(async_context_, password));`
`async_context_->data = this;`
`redisAsyncSetDisconnectCallback(async_context_, RedisAsyncContextDisconnectCallback);`

Reply:
You are right. But I think it's better to be placed between `ConnectWithRetries` and `AuthenticateRedis`. PR updated.